Make additional_headers configurable

DL6ER opened the pull request:
# What does this implement/fix?

Make additional_headers property of the embedded web server CivetWeb editable by users via the new config option `webserver.headers`. The default value of this variable has some best-practice headers.

**Related issue or feature (if applicable):** https://github.com/pi-hole/FTL/issues/2215

**Pull request in [docs](https://github.com/pi-hole/docs) with documentation (if applicable):** N/A

---
**By submitting this pull request, I confirm the following:** 

1. I have read and understood the [contributors guide](https://docs.pi-hole.net/guides/github/contributing/), as well as this entire template. I understand which branch to base my commits and Pull Requests against. 
2. I have commented my proposed changes within the code.
3. I am willing to help maintain this change if there are issues with it later.
4. It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
5. I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))

## Checklist:

- [x] The code change is tested and works locally.
- [x] I based my code and PRs against the repositories `developmental` branch.
- [x] I [signed off](https://docs.pi-hole.net/guides/github/how-to-signoff/) all commits. Pi-hole enforces the [DCO](https://docs.pi-hole.net/guides/github/dco/) for all contributions
- [x] I [signed](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) all my commits. Pi-hole requires signatures to verify authorship
- [x] I have read the above and my PR is ready for review.

yubiuser reviewed (changes requested):
It seems something was wrong with https://github.com/pi-hole/FTL/pull/2207

Despite being on this branch I get a lot of log output which was introduced with the mentioned PR

```
2025-02-21 22:49:54.001 CET [126824/T126940] INFO: NTP server listening on :::123 (IPv6)
2025-02-21 22:51:26.592 CET [126824/T126952] INFO: Local URI: "/admin/"
2025-02-21 22:51:26.643 CET [126824/T126953] INFO: Local URI: "/admin/login"
2025-02-21 22:51:26.724 CET [126824/T126952] INFO: Local URI: "/admin/vendor/fonts/SourceSansPro/SourceSansPro.css"
2025-02-21 22:51:26.724 CET [126824/T126953] INFO: Local URI: "/admin/vendor/nprogress/nprogress.min.css"
2025-02-21 22:51:26.724 CET [126824/T126954] INFO: Local URI: "/admin/vendor/bootstrap/css/bootstrap.min.css"
2025-02-21 22:51:26.762 CET [126824/T126952] INFO: Local URI: "/admin/vendor/animate/animate.min.css"
2025-02-21 22:51:26.763 CET [126824/T126953] INFO: Local URI: "/admin/vendor/bstreeview/bstreeview.min.css"
2025-02-21 22:51:26.764 CET [126824/T126954] INFO: Local URI: "/admin/vendor/select2/select2.min.css"
2025-02-21 22:51:26.797 CET [126824/T126952] INFO: Local URI: "/admin/style/pi-hole.css"
2025-02-21 22:51:26.797 CET [126824/T126953] INFO: Local URI: "/admin/vendor/adminLTE/AdminLTE.min.css"
2025-02-21 22:51:26.806 CET [126824/T126954] INFO: Local URI: "/admin/style/themes/default-dark.css"
2025-02-21 22:51:26.830 CET [126824/T126953] INFO: Local URI: "/admin/vendor/waitMe/waitMe.min.css"
2025-02-21 22:51:26.830 CET [126824/T126952] INFO: Local URI: "/admin/vendor/nprogress/nprogress.min.js"
2025-02-21 22:51:26.840 CET [126824/T126954] INFO: Local URI: "/admin/vendor/jquery/jquery.min.js"
2025-02-21 22:51:26.860 CET [126824/T126952] INFO: Local URI: "/admin/vendor/bootstrap/js/bootstrap.min.js"
2025-02-21 22:51:26.860 CET [126824/T126953] INFO: Local URI: "/admin/vendor/adminLTE/adminlte.min.js"
2025-02-21 22:51:26.872 CET [126824/T126954] INFO: Local URI: "/admin/vendor/bootstrap-notify/bootstrap-notify.min.js"
2025-02-21 22:51:26.889 CET [126824/T126952] INFO: Local URI: "/admin/vendor/font-awesome/all.min.js"
2025-02-21 22:51:26.889 CET [126824/T126953] INFO: Local URI: "/admin/scripts/js/utils.js"
2025-02-21 22:51:26.901 CET [126824/T126954] INFO: Local URI: "/admin/vendor/waitMe/waitMe.min.js"
2025-02-21 22:51:26.918 CET [126824/T126953] INFO: Local URI: "/admin/scripts/js/login.js"
```

yubiuser reviewed (commented):
I tried this branch, but could not get custom headers injected. Instead, even with no configuration change,  'Settings' were unusable and gave the following error

![2025-02-21_23-26](https://github.com/user-attachments/assets/60fa9da3-71d9-4fbb-bded-c9e8f4ae02b5)

DL6ER commented:
![grafik](https://github.com/user-attachments/assets/9ce55ccd-16fe-4be1-a219-cf85cb721eae)

Interesting that it can be `null` immediately after the `if (typeof allowed === object)`...

DL6ER commented:
Rebased on latest `development` and added an automatic check to ensure the CI complains loudly if the `allowed` property of config items needing it is again forgotten in some future PR.

DL6ER commented:
![grafik](https://github.com/user-attachments/assets/789b01fc-c5d9-45ea-8785-b80a65841226)

yubiuser reviewed (commented):
Works, but 2 'issues'

1. It needs a full FTL restart, the 'automatic' restart after the configuration has changed is not enough
2. The moment, one presses 'save&apply' from 'All Settings' things go bad - the web interface does not include/render the `\r\n` which then don't get saved into the `toml` file and break the headers
____

![2025-02-22_11-06](https://github.com/user-attachments/assets/f89c1e1f-2954-4c79-ad6e-309637fdbb27)

DL6ER commented:
The escaping needs to be done in Javascript, I guess. Will look at this later

DL6ER commented:
@yubiuser See my last commit: Instead of a long string, `webserver.headers` is now a list of individual headers. FTL will then take care of stapling them together, inserting the correct line endings itself.

![grafik](https://github.com/user-attachments/assets/4932caed-e8f4-45f8-882f-305724c10e0b)

yubiuser reviewed (commented):
Really good. Works fine now. Last issue: When switching to this branch, the headers are empty by default

```
#     array of HTTP headers
  headers = [] ### CHANGED, default = [ "Content-Security-Policy: default-src 'self' 'unsafe-inline';", "X-Frame-Options: DENY", "X-XSS-Protection: 0", "X-Content-Type-Options: nosniff", "Referrer-Policy: strict-origin-when-cross-origin" ]

```
